balancer.go

```diff
@@ -158,3 +160,98 @@ func (h *Hash) Balance(msg Message, partitions ...int) (partition int) {
 
 	return
 }
+
+type randomBalancer struct {
+	mock int // mocked return value, used for testing
+}
+
+func (b randomBalancer) Balance(msg Message, partitions ...int) (partition int) {
+	if b.mock != 0 {
+		return b.mock
+	}
+	return partitions[rand.Int()%len(partitions)]
+}
+
+// CRC32Balancer is a Balancer that uses the CRC32 hash function to determine which
+// partition to route messages to.  This ensures that messages with the same key
+// are routed to the same partition.  A random partition will be selected if either
+// the key is nil, or an the empty byte slice.
+//
+// This balancer strategy is the default partitioner used by the librdkafka library
+// and the language bindings that are built on top of it, including the
+// github.com/confluentinc/confluent-kafka-go Go package.
+type CRC32Balancer struct {
+	rb randomBalancer
+}
+
+func (b CRC32Balancer) Balance(msg Message, partitions ...int) (partition int) {
+	if len(msg.Key) == 0 {
```

Comment:
Just to double check, does the librdkafka library differentiate between a key that is the empty string and a nil key?  A quick comment here to clarify would be appreciated.

Reply:
Looks like this matches the default `consistent_random` partitioner (empty or nil key is randomly assigned).  There's also a `consistent` algo that does assigns empty or nil keys to a single partition.  It might make sense to decouple the `randomBalancer` so that we could support both modes.

See https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md